pkg/server/server.go

```diff
@@ -94,6 +112,21 @@ func WithCredentials(creds credentials.Credentials) Option {
 	}
 }
 
+func (m *Server) makeConnectArg(hostname string, params hostParams) (string, int) {
+	host := hostname
+	if params.GetIP().IsValid() {
```

Comment:
In the same function we work with both params.GetIP and params.ip, I suggest to unify approach and use pointer to params (as getParams already returns pointer) with GetIP to avoid NPE